### PR TITLE
Add SetNameAndID to Container interface

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

The main name generation of a container is now done in the new Container
interface. Necessary getters and tests have been added as well.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

/cc @rhafer

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM

File: pkg/container/container.go [hunk 1]
```diff
@@ -2,31 +2,46 @@ package container
...
 
 // Container is the main public container interface
 type Container interface {
+	// All set methods are usually called in order of their definition
+
 	// SetConfig sets the configuration to the container and validates it
 	SetConfig(*pb.ContainerConfig) error
 
+	// SetNameAndID retrieves a container name and ID
+	SetNameAndID(*pb.PodSandboxMetadata) error
```
Comment (nalind) [inline pkg/container/container.go]: retrieves or sets?

Comment (saschagrunert) [reply pkg/container/container.go]: Sets for sure, fixed that :)

**Comment by haircommander** [comment]
```

08:00:22 sed: can't read /etc/crio/crio.conf: No such file or directory
08:00:22 Failed at 142: sudo sed -i "/\[crio.runtime\]/a$network_ns_flag = true" "$crio_config_file"
```
I haven't seen this today, maybe it's been fixed?
/retest

**Comment by saschagrunert** [comment]
> ```
> 
> 08:00:22 sed: can't read /etc/crio/crio.conf: No such file or directory
> 08:00:22 Failed at 142: sudo sed -i "/\[crio.runtime\]/a$network_ns_flag = true" "$crio_config_file"
> ```
> 
> I haven't seen this today, maybe it's been fixed?
> /retest

Working on a fix: https://github.com/kata-containers/tests/pull/2401

**Comment by haircommander** [comment]
/retest
provisioning :\

**Pull request merged**